File: The Core/Converter.cpp
#include <iostream>
#include <string>
#include <math.h>

#include "Converter.h"

int Converter::to_dec(std::string hex_number){
	return Details::to_dec_system(hex_number, 16);
}

std::string Converter::to_dec(const std::string &hex_number, size_t number_size){
	return Details::added_zeros(Details::to_k_system(to_dec(hex_number), 10), number_size);
}

std::string Converter::to_hex(int dec_number){
	return Details::to_k_system(dec_number, 16);
}

std::string Converter::to_hex(int dec_number, size_t number_size){
	return Details::added_zeros(to_hex(dec_number), number_size);
}

std::string Converter::to_hex(const std::string &dec_number){
	return to_hex(Details::to_dec_system(dec_number, 10));
}

std::string Converter::to_hex(std::string dec_number, size_t number_size){
	return Details::added_zeros(to_hex(dec_number), number_size);
}

std::string Converter::Details::to_k_system(int dec_number, size_t k){
	return dec_number ? to_k_system(dec_number / k, k) + int_to_char(dec_number %  k) : std::string();
}

int Converter::Details::to_dec_system(const std::string &number, size_t k, size_t cur){
	return cur < number.size() ? char_to_int(number[cur]) * int(pow(double(k), number.size() - 1 - cur)) + 
		to_dec_system(number, k, cur + 1) : 0;
}

char Converter::Details::int_to_char(int number){
	if (number > 36 || number < 0)
		throw std::exception();

	return number > 9 ? number - 10 + 'A' : number + '0';
}

int Converter::Details::char_to_int(char number){
	if (!((number >= '0' && number <= '9') || (number >= 'A' && number <= 'Z')))
		throw std::exception();

	return number <= '9' ? number - '0' : number + 10 - 'A';
}

std::string Converter::Details::added_zeros(const std::string &number, size_t number_size){
	int numer_of_zeros((int)number_size - (int)number.size());

	if (numer_of_zeros < 0)
		throw std::exception();

	return std::string(numer_of_zeros, '0') + number;
}

File: Windows - GUI CS/Converter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LM_GUI_UP
{
    static class Converter
    {
        #region Class Details
        private static class Details
        {
            public static int ToDecSystem (string number, uint k, int cur = 0){
                	return cur < number.Length ? CharToInt(number[cur]) * (int)(Math.Pow(k, number.Length - 1 - cur)) + 
		            ToDecSystem(number, k, cur + 1) : 0;
            }
            private static int CharToInt(char number)
            {
                if (!((number >= '0' && number <= '9') || (number >= 'A' && number <= 'Z')))
		        throw new Exception("");
                //Invalid number
	
	            return number <= '9' ? number - '0' : number + 10 - 'A';
            }
            private static char IntToChar(int number)
            {
                if (number > 36 || number < 0)
                    throw new Exception();
                //invalid numerical system

                return (char)(number > 9 ? number - 10 + 'A' : number + '0');
            }
            public static string AddedZeros(string number, uint numberSize){
                StringBuilder resultingNumber = new StringBuilder();
                int number_of_zeros = (int)numberSize - (int)number.Length;

                if(number_of_zeros < 0)
                    throw new Exception();
                    //Invalid argument

                //possible mistake
                for(int i = 0; i < number_of_zeros;i++)
                    resultingNumber.Append('0');

                resultingNumber.Append(number);

                return resultingNumber.ToString();
            }

            public static string ToKSystem(int decNumber, uint k)
            {
                return decNumber != 0 ? ToKSystem((int)(decNumber / k), k) + IntToChar((int)(decNumber %  k)) : "";
            }

            
        }
        #endregion
        public static int HexToDec(string hexNumber)
        {
            return Details.ToDecSystem(hexNumber, 16);
        }

        public static string DecToHex(int decNumber, uint numberSize)
        {
            return Details.AddedZeros(DecToHex(decNumber), numberSize);
        }

        public static string DecToHex(int decNumber)
        {
            return Details.ToKSystem(decNumber, 16);
        }
    }
}
